src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_6001_7000.cpp

```diff
@@ -5592,7 +5597,75 @@ GUI_TEST_CLASS_DEFINITION(test_6754) {
     GTUtilsMSAEditorSequenceArea::checkSelectedRect(os, QRect(0, 0, 1, 1));
     CHECK_SET_ERR(!l.hasErrors(), "Errors in log: " + l.getJoinedErrorString());
 }
+GUI_TEST_CLASS_DEFINITION(test_6759)
+{
+    GTLogTracer l;
+
+    // The test just check that there are no crash hile rotating circular view
+    //    1. Open sequence
+    //    2. Open annotation file
+    //    3. Add annotation file to sequence
+    //    4. Find splitter
+    //    5. Find scroll in splitter
+    //    6. Scroll to end, then to begin
+
+    GTFileDialog::openFileList(os,
+                               testDir + "_common_data/regression/6759/",
+                               QStringList() << "annotations.gb"
+                                             << "sequence.gb");
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+    CHECK_SET_ERR(GTUtilsProjectTreeView::checkItem(os, "Unknown features"),
+                  "No 'Unknown features' object!");
+
+    //    Use context menu on annotation in tree view
+    GTUtilsDialog::waitForDialog(os,
+                                 new ProjectTreeItemSelectorDialogFiller(os,
+                                                                         "annotations.gb",
+                                                                         "Unknown features"));
+    GTUtilsDialog::waitForDialog(os, new CreateObjectRelationDialogFiller(os));
+    GTUtilsDialog::waitForDialog(os,
+                                 new PopupChooserByText(os, {"Add", "Objects with annotations..."}));
+    //    On question "Found annotations that are out of sequence range, continue?" answer "Yes"
+    GTUtilsDialog::waitForDialog(os, new MessageBoxDialogFiller(os, QMessageBox::Yes));
+
+    GTUtilsSequenceView::openPopupMenuOnSequenceViewArea(os);
+    //    Check {add-> Objects with annotations} action
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+
+    int seqNum = GTUtilsSequenceView::getSeqWidgetsNumber(os);
+    CHECK_SET_ERR(seqNum == 1, QString("Too many seqWidgets count").arg(seqNum));
+
+    QScrollBar *horScroll = GTWidget::findExactWidget<QScrollBar *>(os, "CircularViewSplitter_horScroll");
+    CHECK_SET_ERR(horScroll != nullptr, "Can't find circular view splitter's horScrollbar");
 
+    // We use sleep as scrolling is executing too fast without sleep
+    horScroll->setValue(horScroll->value() + 13);
```

Comment:
The problem remains: do not modify UI state from non-UI threads. This is a 'must' for any QT code (it is not thread safe)